.github/workflows/test.yml

```diff
@@ -0,0 +1,145 @@
+name: Test Workflow
+on:
+  workflow_call:
+
+jobs:
+  unit-tests:
+    name: Unit Tests
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+
+      - name: Setup Java 11
```

Comment:
I think you can consider adding cache (actions/cache@v4) so that it runs a bit faster

Reply:
I don't believe caching will help here as Java versions are already cached and the job step takes less than a second.